### PR TITLE
Add user interactivity

### PR DESCRIPTION
**Pull request by jcmugabi**
**What does this PR do**
- Pagination using infinite_scroll_pagination
- Share functionality
- Searching users by username

**Tasks to be completed**
- Internet state

### CODE REVIEW

**Comment by trishaagaba** [comment]
This is good work

**Review by trishaagaba — COMMENTED** [review]
This is good work

File: lib/data/remote/api_urls.dart
Comment (NabunyaLilian) [inline lib/data/remote/api_urls.dart:1]: remove comments

File: lib/presentation/screens/users_screen.dart
Comment (NabunyaLilian) [inline lib/presentation/screens/users_screen.dart:1]: This method "_checkConnectivity()" in used in different files, you can create a utils folder, then create a helper functions page and add it in there. Then reuse it

File: lib/presentation/state/providers/internet_connection_provider.dart
Comment (NabunyaLilian) [inline lib/presentation/state/providers/internet_connection_provider.dart:1]: Remove this widget in the provider. Providers shouldn't have widgets, just read the boolean isConnected from the widgets and create the card from there. You cn make the card a reusable widget, don't use methods , also use names that portray meaning.

**Review by NabunyaLilian — CHANGES_REQUESTED** [review]
This is good work Mugabi, but resolve the changes requested

File: lib/data/remote/api_urls.dart
Comment (jcmugabi) [reply lib/data/remote/api_urls.dart:1]: Working on changes.

**Review by NabunyaLilian — APPROVED** [review]

**Pull request merged**